cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -800,6 +800,9 @@ Status NdarrayToArrow(MemoryPool* pool, PyObject* ao, PyObject* mo, bool from_pa
   if (!PyArray_Check(ao)) {
     return Status::Invalid("Input object was not a NumPy array");
   }
+  if (PyArray_NDIM(reinterpret_cast<PyArrayObject*>(ao)) != 1) {
+    return Status::Invalid("only handle 1-dimensional arrays");
```

Comment:
`NumPyConverter::Convert` does the check again, but sadly we cannot raise from the constructor, so we'd need a factory to return the `Status`...

Reply:
Actually, that check can probably be removed? (as the NumpyConverter is only used from this function)